feature-libs/checkout/base/root/events/checkout-legacy-store-event.listener.spec.ts

```diff
@@ -0,0 +1,102 @@
+import { TestBed } from '@angular/core/testing';
+import { MockStore, provideMockStore } from '@ngrx/store/testing';
+import { CartActions } from '@spartacus/cart/main/core';
+import { LoadCartEvent, RemoveCartEvent } from '@spartacus/cart/main/root';
+import {
+  createFrom,
+  CxEvent,
+  EventService,
+  LoadUserAddressesEvent,
+  LoadUserPaymentMethodsEvent,
+  UserActions,
+} from '@spartacus/core';
+import { Subject } from 'rxjs';
+import { CheckoutLegacyStoreEventListener } from './checkout-legacy-store-event.listener';
+import createSpy = jasmine.createSpy;
+
+const mockEventStream$ = new Subject<CxEvent>();
+const mockUserId = 'testUserId';
+const mockCartId = 'testCartId';
+
+class MockEventService implements Partial<EventService> {
+  get = createSpy().and.returnValue(mockEventStream$.asObservable());
+  dispatch = createSpy();
+}
+
+describe(`CheckoutPaymentEventListener`, () => {
+  let store: MockStore;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      providers: [
+        CheckoutLegacyStoreEventListener,
+        provideMockStore(),
+        {
+          provide: EventService,
+          useClass: MockEventService,
+        },
+      ],
+    });
+
+    TestBed.inject(CheckoutLegacyStoreEventListener);
+    store = TestBed.inject(MockStore);
+  });
+
+  describe(`onUserAddressAction`, () => {
+    it(`should dispatch UserActions.LoadUserAddresses`, () => {
+      spyOn(store, 'dispatch').and.stub();
```

Comment:
think we can move this under line 42 since all use test cases needs to stub the dispatch

Reply:
yep, nice :)